lib/src/tag/tagger.dart

```diff
@@ -221,22 +221,24 @@ class Tagger {
     List<Explanation> explanations, {
     bool trustDeclarations = true,
   }) {
-    try {
-      if (_isBinaryOnly) {
-        // TODO: consider checking `platforms:` is present in `pubspec.yaml`
-        tags.addAll(Platform.binaryOnlyAssignedPlatforms.map((p) => p.tag));
-        explanations.addAll(Platform.binaryOnlyNotAssignedPlatforms.map((p) =>
-            Explanation(p.name,
-                'Cannot assign ${p.name} automatically to a binary only package.',
-                tag: p.tag)));
-      } else {
-        for (final platform in Platform.recognizedPlatforms) {
-          final extraEnabledLibs = <String>{
-            if (platform.allowsNativeJit && !_usesFlutter)
-              ...Runtime.nativeJit.enabledLibs,
-          };
+    if (_isBinaryOnly) {
+      // TODO: consider checking `platforms:` is present in `pubspec.yaml`
+      tags.addAll(Platform.binaryOnlyAssignedPlatforms.map((p) => p.tag));
+      explanations.addAll(Platform.binaryOnlyNotAssignedPlatforms.map((p) =>
+          Explanation(p.name,
+              'Cannot assign ${p.name} automatically to a binary only package.',
+              tag: p.tag)));
+      return;
+    }
+
+    final platformResults = <TaggingResult>[];
+    for (final platform in Platform.recognizedPlatforms) {
+      TaggingResult checkRuntime(Runtime runtime) {
```

Comment:
Could we move this `checkRuntime` to be a method next to `platformTags`?

Reply:
Done.